Gravity/Gravity/DAL/RSAPI/RsapiDao.Insert.cs

```diff
@@ -124,23 +122,88 @@ public void InsertChildListObjects<T>(IList<T> objectsToInserted, int parentArti
 				var rdosToBeInserted = objectsToInserted.Select(x => x.ToRdo()).ToArray();
 
 				rsapiProvider.Create(rdosToBeInserted);
-			}			
+			}
+		}
+
+		private bool InsertUpdateSingleObjectFields(BaseDto objectToInsert)
+		{
+			foreach (var propertyInfo in objectToInsert.GetType().GetProperties().Where(c =>
+				c.GetCustomAttribute<RelativityObjectFieldAttribute>()?.FieldType == RdoFieldType.SingleObject))
+			{
+				var fieldGuid = propertyInfo.GetCustomAttribute<RelativityObjectFieldAttribute>()?.FieldGuid;
+				var fieldValue = (objectToInsert.GetPropertyValue(propertyInfo.Name) as BaseDto);
+
+				if (fieldGuid != null && fieldValue != null)
+				{
+					//TODO: better test to see if contains value...if all fields are null, not need
```

Comment:
I'd insert it anyways; expected behavior is to insert an object if you ask for it (no different than, say, what would happen with RSAPI today).

Reply:
Good logical thinking as always.  Cleaned it up.